platoon/channel.py

```diff
@@ -49,26 +35,38 @@ class Controller(object):
     ----------
     port : int
         The port number to communicate over
-    cport : int
+    control_port : int
         The control port number.
     hwm : int
         High water mark (see pyzmq docs).
 
     """
 
-    def __init__(self, port=None, cport=None, hwm=10):
+    def __init__(self, control_port, port=None, hwm=10):
 
         self._should_stop = False
         self._worker_list = set()
+        self._need_init = True
 
         if port:
             self.init_data(port, hwm)
-        if cport:
-            self.init_control(cport)
+
+        self._init_control_socket(control_port)
+
+        ## Cleanup and init global lock and job_uid name ##
+        self._job_uid = "platoon_{0}_{1}".format(os.path.basename(os.path.expanduser('~')), control_port)
+
+        # The ExistentialError is apparently the only way to verify if the semaphore/shared_memory exists.
+        try:
+            posix_ipc.unlink_semaphore(self._job_uid+"lock")
+        except posix_ipc.ExistentialError:
+            pass
+        # Initializing lock
+        posix_ipc.Semaphore(self._job_uid+"lock", posix_ipc.O_CREAT, initial_value=1)
```

Comment:
This change was done to allow starting all worker at the same time. I don't see problems with that and not doing this would make thing more complicated elsewhere I think.